### PR TITLE
Use C++14 helper types for traits

### PR DESCRIPTION
**Pull request by sbastrakov**
Replace the old `typename std::[trait]<...>::type` with C++14 `std::[trait]_t<...>`.
Add occasionally missing `#include <type_traits>`.

There is a similar helper for values, but this is C++17, so not changed.

TODO: test properly, since  I could have introduced typos.

### CODE REVIEW

File: test/common/include/alpaka/test/mem/view/Iterator.hpp [hunk 1]
```diff
@@ -35,10 +37,10 @@ namespace alpaka
                     template<
                         typename T,
                         typename TSource>
-                    using MimicConst = typename std::conditional<
+                    using MimicConst = std::conditional_t<
                         std::is_const<TSource>::value,
-                        typename std::add_const<T>::type,
-                        typename std::remove_const<T>::type>;
+                        std::add_const_t<T>,
+                        std::remove_const_t<T>>;
```
Comment (BenjaminW3) [inline test/common/include/alpaka/test/mem/view/Iterator.hpp:43]: This is an alias template. You probably have to adjust the call to `typename MimicConst<T, TSource>::type` as well now to make the compiler happy.

**Comment by BenjaminW3** [comment]
Thanks for modernizing the code!

File: test/common/include/alpaka/test/mem/view/Iterator.hpp [hunk 2]
```diff
@@ -50,7 +52,7 @@ namespace alpaka
                         typename TSfinae = void>
                     class IteratorView
                     {
-                        using TViewDecayed = typename std::decay<TView>::type;
+                        using TViewDecayed = std::decay_t<TView>;
                         using Dim = alpaka::dim::Dim<TViewDecayed>;
                         using Idx = alpaka::idx::Idx<TViewDecayed>;
                         using Elem = typename MimicConst<alpaka::elem::Elem<TViewDecayed>, TView>::type;
```
Comment (BenjaminW3) [inline test/common/include/alpaka/test/mem/view/Iterator.hpp]: here ;-)
and further down in this file.

Comment (sbastrakov) [reply test/common/include/alpaka/test/mem/view/Iterator.hpp]: Thanks! Fixed and rebased.

**Review by BenjaminW3 — APPROVED** [review]

**Pull request merged**